test/helpers/query.cast$expr.test.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const { Schema } = require('../common').mongoose;
+const assert = require('assert');
+const cast$expr = require('../../lib/helpers/query/cast$expr');
+
+describe('castexpr', function() {
+  it('casts comparisons', function() {
```

Comment:
Would be nice if we could support `$cond: [myCondition, trueCase, falseCase]` as well.

Would happily work on that in a next PR or on this one. @vkarpov15